metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -1394,7 +1394,9 @@ def get_connection_object_page(
         `fetch_size:` The number of records to retrieve from Tableau
             Server in a single API call, starting from the current cursor position on Tableau Server.
         """
-        retries_remaining = retries_remaining or self.config.max_retries
+        retries_remaining = (
+            self.config.max_retries if retries_remaining is None else retries_remaining
+        )
```

Comment:
the problem was 0 is _falsy_ and so reset to max_retries, right? nice catch!

Reply:
yes, exactly